aptos-move/block-executor/src/scheduler_status.rs

```diff
@@ -0,0 +1,1387 @@
+   - When the scheduler selects a transaction, it transitions the status to `Executing` via
+     the `try_start_executing` method.
+
+2. Abortion Process:
+   - A transaction incarnation may be aborted if it reads data that is later modified in a way
+     that would cause the transaction to read different values if it executed again. This
+     signals the need for re-execution with an incremented incarnation number.
+   - In BlockSTMv2, a transaction can be aborted while executing or after execution finishes.
+   - The abortion happens in two distinct phases:
+
+   a) Try Abort Phase:
+      - `try_abort` is called with an incarnation number and succeeds if the incarnation has
+        started executing and has not already been aborted.
+      - This serves as an efficient test-and-set filter for multiple abort attempts (which
+        can occur when a transaction makes multiple reads that may each be invalidated by
+        different transactions).
+      - Early detection allows the ongoing execution to stop immediately rather than continue
+        work that will ultimately be discarded.
+
+   b) Finish Abort Phase:
+      - A successful `try_abort` must be followed by a `finish_abort` call.
+      - When transaction T1 successfully aborts transaction T2 (where T2 > T1):
+        • T2 stops executing as soon as possible
+        • Subsequent scheduling of T2 may wait until T1 finishes, since T1 has higher
+          priority (lower index)
+        • After T1 completes, the worker can process all related aborts in batch, including
+          calling `finish_abort`, tracking dependencies, and propagating stalls
+
+3. Execution Completion:
+   - When execution finishes, `finish_execution` is called on the status.
+   - If the status was `Aborted`, it transitions to `PendingScheduling` for the next incarnation.
+   - If the status was `Executing`, it transitions to `Executed`.
+
+Status Transition Diagram:
+
+PendingScheduling(i)
+    |
+    | try_start_executing
+    |
+    ↓                       finish_execution
+Executing(i) ------------------------------> Executed(i)
+    |                                           |
+    | try_abort(i) + finish_abort(i)            | try_abort(i) + finish_abort(i)
+    |                                           |
+    ↓                    finish_execution       ↓
+Aborted(i) ------------------------------> PendingScheduling(i+1)
+
+Note: `try_abort` doesn't change the status directly but marks the transaction for
+abortion. The actual status change occurs during `finish_abort`. Both steps are
+required for the complete abortion process.
+
+============================== Transaction Stall Mechanism ==============================
+
+In the BlockSTMv2 scheduler, a transaction status can be "stalled," meaning there have been
+more `add_stall` than `remove_stall` calls on its status. Each successful `add_stall` call
+requires the caller to eventually perform a corresponding `remove_stall`.
```

Comment:
"the caller" -> has to be the same caller?

Reply:
great point, it doesn't. updated.